### DISCUSSION TITLE
Is Chunks Cache Required if Running Blocks Storage Engine?

### DISCUSSION THREAD

**Discussion by michael-burt** [cortexproject/cortex#4565] [Q&A] [answered]
Is the Chunks Cache, [documented here](https://cortexmetrics.io/docs/chunks-storage/caching/#chunks-cache), a component that should be deployed when using the Blocks storage engine?

This page includes documentation for Index Read, Index Write, Query Results, & Chunks Caches. Are all of these caches irrelevant for Blocks storage engine? I see both the Index Read & Chunks Cache on the Blocks Storage architecture diagram [here](https://cortexmetrics.io/docs/blocks-storage/#architecture).

**Comment by dgonzalezruiz** [comment, marked as answer]
Yes, although the wording can be confusing. From https://cortexmetrics.io/docs/blocks-storage/ --> 
```
). Each block is composed by chunk files - containing the timestamp-value pairs for multiple series - and an index, which indexes metric names and labels to time series in the chunk files.

```
So, for caching your blocks storage query path / avoiding to refetch already queried samples, you would have to at least run both `chunk` and `index` cache.
You also can configure a separate cache for metadata & query results for even more caching, and there are some more tips for scaling it on https://cortexmetrics.io/docs/blocks-storage/production-tips/#caching

**Reply by michael-burt** [reply]
Thanks for the explanation @dgonzalezruiz
